### PR TITLE
Kpi#2917 django api compression

### PR DESCRIPTION
**Pull request by JacquelineMorrissette**
## Checklist
1. [x] If you've added code that should be tested, add tests
-No code written but moved and cleaned up nginx config files.
2. [n/a] If you've changed APIs, update (or create) the documentation
3. [x] Ensure the tests pass
4. [x] Make sure that your code lints and that you've followed [our coding style]
5. [x] Write a description of your work suitable for publishing on [our forum]
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)

## Description

Gzip is now enabled on the server for all API responses.
This means a decrease in the time it takes to request data and have the response returned to the user.

Removed parameters that are no longer used for production or development.

## Related Issues
Closes kobotoolbox/kpi#2917

### CODE REVIEW

File: nginx/docker-entrypoint.d/30-init-kobo-nginx.sh [hunk 1]
```diff
@@ -10,26 +10,6 @@ KOBOCAT_PRODUCTION_LOCATION_STATIC='location /static {
 
 KPI_PRODUCTION_LOCATION_STATIC='location /static {
         alias /srv/www/kpi;
```
Comment (noliveleger) [inline nginx/docker-entrypoint.d/30-init-kobo-nginx.sh:12]: `KPI_PRODUCTION_LOCATION_STATIC` and `KOBOCAT_PRODUCTION_LOCATION_STATIC` are useless now. 
You could remove both of them and replace `${kobocat_location_static}` and `${kpi_location_static}` in `nginx_site_default.conf.tmpl` with the corresponding `location block`.

File: nginx/docker-entrypoint.d/30-init-kobo-nginx.sh [hunk 2]
```diff
@@ -98,21 +78,6 @@ for container_name in "${!container_ports[@]}"; do
...
-    elif [[ "${static_files_server^^}" == "DJANGO" && "${django_debug^^}" == "TRUE" ]]; then
-        echo "Serving static files for container ${container_name} from Django."
-        location_static=''
-    elif [[ "${static_files_server^^}" == "DJANGO" && "${django_debug^^}" != "TRUE" ]]; then
-        echo "Cannot serve static files from Django for container \`${container_name}\` unless \`${django_debug_varname}\` set to \"True\" in \`/envfiles/${container_name}.txt\`."
-        exit 1
-    fi
     location_static_varname="${container_name}_location_static"
     export ${location_static_varname}="${location_static}"
     templated_var_refs+=" \${$location_static_varname}}"
```
Comment (noliveleger) [inline nginx/docker-entrypoint.d/30-init-kobo-nginx.sh:83]: I think we don't need this block anymore too. If so, please remove it.

File: nginx/kobo-docker-scripts/nginx.conf [hunk 3]
```diff
@@ -50,8 +50,25 @@ http {
...
-    gzip_disable "msie6";
+        # gzip configs from here
+        # http://stackoverflow.com/a/12644530/3088435
+        gzip on;
+        gzip_disable "msie6";
+        gzip_comp_level 6;
+        gzip_min_length 1100;
+        gzip_buffers 16 8k;
+        gzip_proxied any;
+        gzip_types
```
Comment (noliveleger) [inline nginx/kobo-docker-scripts/nginx.conf:61]: Let's keep the same indentation.

File: nginx/kobo-docker-scripts/nginx.conf [hunk 4]
```diff
@@ -50,8 +50,25 @@ http {
...
+            text/plain
+            text/css
+            text/js
+            text/xml
+            text/javascript
+            application/javascript
+            application/x-javascript
+            application/json
+            application/xml
+            application/xml+rss;
```
Comment (noliveleger) [inline nginx/kobo-docker-scripts/nginx.conf:71]: As I suggested, we should add svg too  (i.e. `image/svg+xml`)

**Review by noliveleger — CHANGES_REQUESTED** [review]

File: nginx/kobo-docker-scripts/nginx.conf (hunk 4 above)
Comment (JacquelineMorrissette) [reply nginx/kobo-docker-scripts/nginx.conf]: I added image/svg+xml to the `gzip_types` list

File: nginx/docker-entrypoint.d/30-init-kobo-nginx.sh (hunk 2 above)
Comment (JacquelineMorrissette) [reply nginx/docker-entrypoint.d/30-init-kobo-nginx.sh]: @noliveleger I've done this with the latest [commit](https://github.com/kobotoolbox/kobo-docker/pull/312/commits/374a967ea253bb3afda969cfd2b6cf9f55cf1352), could you verify it for me?

**Review by noliveleger — APPROVED** [review]

**Pull request merged**